Removed iaf_neuron from docstring and documentation examples

stinebuu opened the pull request:
Changed `iaf_neuron` to `iaf_psc_alpha` in docstring examples and in documentation/examples. See #558 for a list. Also made sure PyNEST examples do not show deprecation warnings. This should fix #558. 

topology/doc/user_manual_scripts/* is not changed, as this is done by #621 . 

As discussed in #580, SLI examples are mostly left alone, though I have replaced `iaf_neuron` with `iaf_psc_alpha`.

Silmathoron commented on doc/model_details/noise_generator.ipynb:
```diff
@@ -267,7 +267,7 @@
    "source": [
     "def noise_params(V_mean, V_std, dt=1.0, tau_m=10., C_m=250.):\n",
-    "    'Returns mean and std for noise generator for parameters provided; defaults for iaf_neuron.'\n",
+    "    'Returns mean and std for noise generator for parameters provided.'\n",
```
Maybe leaving the explanation about the fact that ``tau_m=10., C_m=250.`` are the default values for the ``iaf_psc_alpha`` model could be useful...

Silmathoron commented on examples/nest/tsodyks_depressing.sli:
```diff
@@ -21,7 +21,7 @@
 
  /* BeginDocumentation
-Name: test_tsodyks_depressing - sli script for overall test of iaf_neuron model
+Name: test_tsodyks_depressing - sli script for overall test of iaf_psc_exp model
```
Why is it not ``iaf_psc_alpha``? (though I guess this should not be important)

stinebuu replied:
The example already uses `iaf_psc_exp`, so I figured someone had changed the model and forgotten to update the comment. I did change it to `iaf_psc_alpha` first, and got confused when I couldn't find any `ìaf_neuron`s in the actual example. I had the same reasoning with the example below, tsdoyks_facilitating.sli.

heplesser replied:
Actually, this example (and pretty much the other three tsodyks_* examples in examples/nest) are not in good shape. I suggest we remove all four sli examples here, we have proper Python examples for Tsodyks-style facilitation and depression.

Silmathoron commented on examples/nest/tsodyks_facilitating.sli:
```diff
@@ -21,7 +21,7 @@
 
  /* BeginDocumentation
-Name: tsodyks_facilitating - sli script for overall test of iaf_neuron model
+Name: tsodyks_facilitating - sli script for overall test of iaf_psc_exp model
```
see above

heplesser replied:
See above.

Silmathoron commented on nestkernel/nestmodule.cpp:
```diff
@@ -905,95 +905,97 @@ NestModule::MemoryInfoFunction::execute( SLIInterpreter* i ) const
-   SLI [3] [0] 2 PrintNetwork
-   +-[0] Subnet Dim=[12]
+      +- [1] iaf_psc_alpha
+      +- [2]...[11] iaf_cond_neuron
```
I think this should be ``iaf_cond_alpha``

stinebuu replied:
I must have been very focused on `iaf_psc_alpha`! Thanks for the review, I will look at it now.

Silmathoron commented on nestkernel/nestmodule.cpp:
```diff
@@ -905,95 +905,97 @@ NestModule::MemoryInfoFunction::execute( SLIInterpreter* i ) const
-      +- lifb_cond_neuron [2]..(10)..[11]
-      +-[12] Subnet Dim=[2 5 6]
+      +- [1] iaf_psc_alpha
+      +- [2]...[11] iaf_cond_neuron
```
see above

Silmathoron commented on nestkernel/nestmodule.cpp:
```diff
@@ -905,95 +905,97 @@ NestModule::MemoryInfoFunction::execute( SLIInterpreter* i ) const
-      +- lifb_cond_neuron [2]..(10)..[11]
-      +-[12] Subnet Dim=[2 5 6]
+      +- [1] iaf_psc_alpha
+      +- [2]...[11] iaf_cond_neuron
```
see above

Silmathoron commented on nestkernel/nestmodule.cpp:
```diff
@@ -905,95 +905,97 @@ NestModule::MemoryInfoFunction::execute( SLIInterpreter* i ) const
-      +- lifb_cond_neuron [2]..(10)..[11]
-      +-[12] Subnet Dim=[2 5 6]
+      +- [1] iaf_psc_alpha
+      +- [2]...[11] iaf_cond_neuron
```
see above

Silmathoron commented on pynest/examples/gap_junctions_inhibitory_network.py:
```diff
@@ -126,7 +126,7 @@
 `one_to_one` connections.
 """
-n_connection = n_neuron * gap_per_neuron / 2
+n_connection = n_neuron * gap_per_neuron // 2
```
I guess you want to make sure an int is returned even if ``gap_per_neuron`` and ``n_neuron`` are odd... But then I think using ``int(n_neuron * gap_per_neuron / 2)`` might be better since it also works if ``gap_per_neuron`` is a float...

Silmathoron commented on pynest/examples/plot_quantal_stp_synapse.py:
```diff
@@ -98,17 +98,19 @@
-    pylab.plot(vm[t][:t_plot], color='gray', lw=0.5)
-pylab.plot(vm_mean[:t_plot], color='black', lw=2.)
-pylab.plot(vm_reference[0][:t_plot], color='red', lw=2.)
+    pylab.plot(vm[t][:int(t_plot)], color='gray', lw=0.5)
```
I would do the same as for ``t_tot``: ``t_plot = int(t_plot)``

Silmathoron commented on topology/examples/conncon_sources.py:
```diff
@@ -22,7 +22,7 @@
 NEST Topology Module Example
 
-Create two 30x30 layers of iaf_neurons,
+Create two 30x30 layers of iaf_psc_alphas,
```
iaf_psc_alpha neurons

Silmathoron commented on topology/examples/conncon_targets.py:
```diff
@@ -22,7 +22,7 @@
 NEST Topology Module Example
 
-Create two 30x30 layers of iaf_neurons,
+Create two 30x30 layers of iaf_psc_alphas,
```
see above

Silmathoron commented on topology/examples/connex_ew.py:
```diff
@@ -22,7 +22,7 @@
 NEST Topology Module Example
 
-Create two 30x30 layers of iaf_neurons with edge_wrap,
+Create two 30x30 layers of iaf_psc_alphas with edge_wrap,
```
see above

Silmathoron commented on topology/examples/grid_iaf.py:
```diff
@@ -22,7 +22,7 @@
 NEST Topology Module Example
 
-Create layer of 4x3 iaf_neurons, visualize
+Create layer of 4x3 iaf_psc_alphas, visualize
```
see above

Silmathoron commented on topology/examples/grid_iaf_irr.py:
```diff
@@ -22,7 +22,7 @@
 NEST Topology Module Example
 
-Create layer of 12 freely placed iaf_neurons, visualize
+Create layer of 12 freely placed iaf_psc_alphas, visualize
```
see above

Silmathoron commented on topology/examples/grid_iaf_oc.py:
```diff
@@ -22,7 +22,7 @@
 NEST Topology Module Example
 
-Create three layers of 4x3 iaf_neurons, each with different center.
+Create three layers of 4x3 iaf_psc_alphas, each with different center.
```
see above

stinebuu commented:
@heplesser I have now removed the tsodyks sli examples, removed `SuppressedDeprecationWarning` from the Potjans Python example, and added comments saying a deprecation warning will be issued. As in #580 I added a comment in the docstring as well. GetNodes appear in network.py, but to run the example you run example.py, so I added the comment in both scripts docstring.

heplesser commented:
@stinebuu Thank you! The checks in Travis failed due to a Travis problem. I have restarted the Travis builds. I also removed one extraneous comma from a comment using the online editor.
